### DISCUSSION TITLE
Have confirmation for run

### DISCUSSION THREAD

**Discussion by pradip-allwyn** [jhuckaby/Cronicle#771] [Ideas]
Hi, i am testing Cronicle,  I click the run button instead of edit at time, this would be catastrophic on production, Can we have a confirmation dialog to execute the run.

**Comment by jhuckaby** [comment]
I'll add this in the next release, thank you.  I have to make it optional and opt-in, however, as I suspect many people won't want it.

**Comment by jhuckaby** [comment]
Okay, in [v0.9.52](https://github.com/jhuckaby/Cronicle/releases/tag/v0.9.52) which was just released, you can now set the following configuration property in your `/opt/cronicle/conf/config.json` file, but please place it *inside* the `client` object:

```js
"prompt_before_run": true
```

Example `client` object with all the other default properties:

```js
"client": {
	"name": "Cronicle",
	"debug": 1,
	
	"prompt_before_run": true,
	
	"default_password_type": "password",
	"privilege_list": [
		{ "id": "admin", "title": "Administrator" },
		{ "id": "create_events", "title": "Create Events" },
		{ "id": "edit_events", "title": "Edit Events" },
		{ "id": "delete_events", "title": "Delete Events" },
		{ "id": "run_events", "title": "Run Events" },
		{ "id": "abort_events", "title": "Abort Events" },
		{ "id": "state_update", "title": "Toggle Scheduler" }
	],
	"new_event_template": { 
		"enabled": 1, 
		"params": {}, 
		"timing": { "minutes": [0] },
		"max_children": 1,
		"timeout": 3600,
		"catch_up": 0,
		"queue_max": 1000
	}
},
```

You should not have to restart Cronicle for this to take effect, just wait about 10 seconds or more, then refresh your browser.

**Reply by pradip-allwyn** [reply]
@jhuckaby this does not seem to work, Can you check this?

**Comment by jhuckaby** [comment]
Works great:

![Screenshot](https://pixlcore.com/screenshots/Screenshot_2024-08-29_at_9.11.31_AM.png)

- Please make sure you are running the very latest version of Cronicle (v0.9.59+).  Upgrade instructions are here: https://github.com/jhuckaby/Cronicle/blob/master/docs/CommandLine.md#upgrading-cronicle
- Please make sure you add the `prompt_before_run` property INSIDE the `client` object, in your `config.json` file.
